### PR TITLE
doc: move standalone documents to existing guides

### PR DESCRIPTION
**Pull request by nashif**
- Move 'getting help' into contrib guidelines
- Reorder sections of contrib guidelines based on content
- Move environment variable document into application development.

This is one other small step for modularizing the docs and general content cleanup. Doing this in small steps to avoid too many changes that are impossible to review or maintain.

Related:
- #34317
- #34286

### CODE REVIEW

**Review by gmarull — APPROVED** [review]

File: doc/application/index.rst [hunk 1]
```diff
@@ -235,6 +235,133 @@ Follow these steps to create a new application directory. (Refer to
    ``source zephyr-env.sh`` or execute ``zephyr-env.cmd`` before building the
    application.
 
+
+.. _env_vars:
+
+Setting Variables
```
Comment (carlescufi) [inline doc/application/index.rst:241]: I am a bit unsure of moving the env vars section to the application development guide, where is the duplication you mention in the commit message?

Comment (nashif) [reply doc/application/index.rst:241]: `Important Build System Variables` for example.

Comment (nashif) [reply doc/application/index.rst:241]: > I am a bit unsure of moving the env vars section to the application development guide,

we can put somewhere else, but this does not need its own standalone page.

**Review by carlescufi — APPROVED** [review]

File: doc/application/index.rst (hunk 1 above)
Comment (mbolivar-nordic) [reply doc/application/index.rst:241]: > Important Build System Variables for example.

I just noticed this PR and I really disagree. I wrote the environment variable page and this change is confusing and a step backwards.  "Important build system variables" is a list of **cmake** variables, **not** environment variables. You are obscuring this important difference by stuffing a section simply titled "Setting variables" in this page. I'll send a logical revert.

**Pull request merged**